parser/src/main/scala/org/globalnames/parser/FactoryAST.scala

```diff
@@ -24,13 +34,19 @@ object FactoryAST {
            comparison: Option[NodeMeta[Comparison]] = None,
            approximation: Option[NodeMeta[Approximation]] = None,
            ignored: Option[String] = None,
-           genusParsed: Boolean = false): NodeMeta[Name] = {
+           genusParsed: Boolean = false)(implicit input: ParserInput): NodeMeta[Name] = {
+    val genus = input.sliceString(uninomial.node.pos.start, uninomial.node.pos.end)
+    val bacteria = bacteriaGenera.contains(genus)
     val name = Name(uninomial.node, subgenus.map { _.node }, species.map { _.node },
                     infraspecies.map { _.node }, comparison.map { _.node },
-                    approximation.map { _.node }, ignored, genusParsed)
+                    approximation.map { _.node }, ignored, bacteria, genusParsed)
+    val bacteriaHomonymWarning = bacteriaHomonymsGenera.contains(genus).option {
+      Vector(Warning(2, "Genus is bacteria name homonym", name))
```

Comment:
this warning should not decrease quality, so the level would be 0. The value could be `The genus is a homonym of a bacterial genus`